pkg/database/database.go

```diff
@@ -62,19 +37,33 @@ type DBInterface interface {
 }
 
 type Database struct {
-	db          *sqlx.DB
-	info        DBInfoInterface
-	paramParser DBParamParser
+	DB       *sqlx.DB
+	Flavor   sqlbuilder.Flavor
+	Selector facade.DBSelector
+	Filter   facade.DBFilter
+	Sorter   facade.DBSorter
+	Inserter facade.DBInserter
+	Deleter  facade.DBDeleter
 }
 
 func NewDatabase() (DBInterface, error) {
 	env, err := newDatabaseEnvironment()
 	if err != nil {
 		return nil, err
 	}
+
+	var creator facade.DBCreator
+	if c, ok := RegisteredDBCreators[env[DbKindEnvVar]]; ok {
+		creator = c(env)
+	} else {
+		panic(fmt.Sprintf("No database registered with name %s", env[DbKindEnvVar]))
```

Comment:
We should return `nil, fmt.Errorf("no database registered with name '%s'", env[DbKindEnvVar])` instead.

Reply:
Updated